app/server.js

```diff
@@ -1,29 +1,79 @@
 const express = require('express')
 const cors = require('cors')
 const pinoHttp = require('pino-http')
-const { PORT } = require('./env')
+const { initialize } = require('express-openapi')
+const swaggerUi = require('swagger-ui-express')
+const path = require('path')
+const bodyParser = require('body-parser')
+const compression = require('compression')
+const { PORT, API_MAJOR_VERSION } = require('./env')
 const logger = require('./logger')
-const router = require('./routes')
-const health = require('./routes/health')
-const auth = require('./routes/auth')
 const checkJwt = require('./auth')
+const v2ApiDoc = require('./api-v2/api-doc')
+const v2ApiService = require('./api-v2/services/apiService')
+const { API_VERSION } = require('./env')
 
 async function createHttpServer() {
   const requestLogger = pinoHttp({ logger })
   const app = express()
 
-  app.use(express.json())
   app.use(cors())
+  app.use(compression())
+  app.use(bodyParser.json())
+
+  app.get('/health', async (req, res) => {
+    res.status(200).send({ version: API_VERSION, status: 'ok' })
+    return
+  })
+
   app.use((req, res, next) => {
     if (req.path !== '/health') {
       requestLogger(req, res)
     }
+
+    if (
+      !req.path.match(`/(${API_MAJOR_VERSION}/api-docs)`) &&
+      !req.path.match(`/(${API_MAJOR_VERSION}/swagger)`) &&
+      !req.path.match(`/(${API_MAJOR_VERSION}/auth)`)
+    ) {
+      return checkJwt(req, res, next)
+    }
+
```

Comment:
This is a little ugly and I think we can do better. Can we not define a `SecurityHandler` for the paths, or is there some reason not to that I'm forgetting?

Reply:
`checkJwt` seems to require `next` to be passed as an argument and the `securityHandlers` `bearerAuth` openApi passes `req` but not `next` so it blows up.

Was going to give you a shout after making the other changes...the Swagger needs the securityHandlers for authentication to work as expected in the UI.